pkg/networking/security_group_resolver.go

```diff
@@ -83,10 +102,31 @@ func (r *defaultSecurityGroupResolver) resolveViaGroupName(ctx context.Context,
 			},
 		},
 	}
+
 	sgs, err := r.ec2Client.DescribeSecurityGroupsAsList(ctx, req)
 	if err != nil {
 		return nil, err
 	}
+
+	resolvedSGNames := make([]string, 0, len(sgs))
+	for _, sg := range sgs {
```

Comment:
we need to add nil check for the `sgs` and `sg.Tags`

Reply:
I think the Golang specification tells us that these would be unnecessarily redundant.

> A "for" statement with a "range" clause iterates through all entries of an array, slice, string or map, values received on a channel, or integer values from zero to an upper limit [[Go 1.22](https://go.dev/ref/spec#Go_1.22)]. For each entry it assigns iteration values to corresponding iteration variables if present and then executes the block. 
> ...
> For an array, pointer to array, or slice value a, the index iteration  [...] _**For a nil slice, the number of iterations is 0.**_

Source: https://go.dev/ref/spec#For_range

Example: https://go.dev/play/p/vBe3S9lbhvh

That said, if you feel strongly about having them or is a common pattern in the codebase I am more than happy to add an extra commit with the additional `nil` checks 😉

Reply:
yeah for the `sgs` it was fine, but how about `sg.Tags`? I know in this function we are filtering by tags, so the returned sgs should have `Tags` field, but just in case if some code changes in the future and we miss this, it will error out because the controller does not check whether there's a Tag spec.

Reply:
> but how about sg.Tags?

I think for `sg.Tags` is the same. If the code changes and the `SecurityGroup` struct removes/renames the `Tags` field, that will produce a compiler error and, hence, we will detect it immediately. Or am I missing something?